plugins/pushbullet.js

```diff
@@ -1,88 +1,186 @@
+    if (config.pushbullet.startMessage) {
+      var subject = this.title + ' Started!'
+      var body = "Gekko has started and is trading LIVE on the " + this.exchange.toUpperCase() + " " + this.asset + "/" + this.currency + ' market using the strategy [' + this.strategy + ']';
+      this.notify(subject, body);
+    } else {
+      log.debug('Pushbullet Notifications Started, not sending startup push.');
+    }
+  };
+  setupPushBullet.call(this)
 };
 
 Pushbullet.prototype.processCandle = function(candle, done) {
-    this.price = candle.close;
+  this.price = candle.close;
+  this.dates.end = candle.start;
 
-    done();
-};
+  if(!this.dates.start) {
+    this.dates.start = candle.start;
+    this.startPrice = candle.close;
+  }
 
-Pushbullet.prototype.processAdvice = function(advice) {
-	if (advice.recommendation == "soft" && pushbulletConfig.muteSoft) return;
+  this.endPrice = candle.close;
 
-	var text = [
-        'Gekko is watching ',
-        config.watch.exchange,
-        ' and has detected a new trend, advice is to go ',
-        advice.recommendation,
-        '.\n\nThe current ',
-        config.watch.asset,
-        ' price is ',
-        this.price
-    ].join('');
+  done();
+}
 
-    var subject = pushbulletConfig.tag+' New advice: go ' + advice.recommendation;
+Pushbullet.prototype.processPortfolioUpdate = function(portfolio) {
+  this.start = portfolio;
+  this.current = _.clone(portfolio);
+}
 
-    this.mail(subject, text);
-};
+Pushbullet.prototype.processTrade = function(trade) {
+  this.trades++;
+  this.current = trade.portfolio;
 
-Pushbullet.prototype.mail = function(subject, content, done) {
-    var pusher = new pushbullet(pushbulletConfig.key);
-    pusher.note(pushbulletConfig.email, subject, content, function(error, response) {
-        if(error || !response) {
-            log.error('Pushbullet ERROR:', error)
-        } else if(response && response.active){
-            log.info('Pushbullet Message Sent')
-        }
-    });
-};
+  var balance = this.current.currency + (trade.price * this.current.asset);
+
+  // don't notify multiple times on trades from the same side
+  // this happens when one advice results in multiple partial trades at different prices
+  if(trade.action !== this.prevAction) {
+    var subject = this.title + ' New Trade: ' + trade.action.toUpperCase() + ' on ' + this.exchange.toUpperCase();
+    var body = trade.action.toUpperCase() + ' ' + this.asset + ' at ' + trade.price.toFixed(this.digitsCurrency) + ' ' +  this.currency;
+    body += '\nBalances:';
+    body += '\n[' + this.currency + ']: ' + this.current.currency.toFixed(this.digitsCurrency); 
+    body += '\n[' + this.asset + ']: ' + this.current.asset.toFixed(this.digitsAsset);
+    body += '\nTotal Value: ' + balance.toFixed(this.digitsCurrency);
+    this.notify(subject, body);
+    this.prevAction = trade.action;
+    this.logRoundtripPart(trade);
+  }
+}
+
+Pushbullet.prototype.logRoundtripPart = function(trade) {
```

Comment:
This code is recalculating everything that is being calculated by this plugin: https://github.com/askmike/gekko/blob/develop/plugins/performanceAnalyzer/performanceAnalyzer.js

If roundtrip information is needed by this plugin it should be exposed by the plugin above and fed into this one.

Reply:
Agree, but I wanted this information now, so rather than start to introduce big changes to performanceAnalyzer I re-did the work here.